src/include/daos/drpc.h

```diff
@@ -27,6 +27,11 @@
 #include <daos/drpc.pb-c.h>
 #include <daos/common.h>
 
+#define DRPC_FREE(ptr)			D_FREE(ptr)
+#define DRPC_ALLOC(ptr, size)		D_ALLOC(ptr, size)
+#define DRPC_ALLOC_PTR(ptr)		D_ALLOC_PTR(ptr)
+#define DRPC_ALLOC_ARRAY(ptr, count)	D_ALLOC_ARRAY(ptr, count)
+
```

Comment:
Again, what value does this add?  It makes it more complex, adds additional macros and yet the backend is both exactly the same, and for reasons I've already stated has to be exactly the same.

Reply:
The idea is just distinguish between one-way allocations (which allocated in our code but freed in 3rd party code).

Reply:
The fault injection code currently passes, so these aren't "one way" allocations, the reason drpc uses the custom allocator is there were a bunch of allocations made by D_ALLOC but freed in drpc or the other way.  For the code that we run though the fault injection framework all these are accounted for.

Of course we don't do allocation tracing on all the code, or all the tests.  Until the server is fixed to be able to shutdown properly and therefore release it's allocations there's not a lot of benefit, but when that bug is fixed I'd like to turn D_ALLOC logging for the entire test suite and do allocation tracing for *all* tests.

Anyway, as it stands all the areas we test are correct, and we're extending the coverage and fixing issues as we find them.